docs/Developer-guidlines.md

```diff
@@ -0,0 +1,109 @@
+contributions, where users fork the main BMI (ims) repo, push changes
+to their personal fork and then create a Pull Request to merge it
+into the master branch of the BMI repository.
+
+### Prior to the pull request
+
+This summarizes what should be done prior to a pull request:
+
+- [ ] If functionality could have architectural implications or controversial, have a discussion with the team. Ideally, prior to coding to save effort.
+- [ ] Ensure any user, deployer or developer documentation is updated.
+- [ ] If a change affects an external API, be sure to update docs/rest\_api.md.
+- [ ] Ensure tests pass (coming soon!) after making your changes.
+- [ ] Ensure that your code is pep8 compliant. 
+
+#### Get agreement from the BMI team
+
+The BMI project appreciates all ideas and submissions. In the past, we've
+discussed several alternatives to how things currently work (which we're trying
+to get better about writing down), and it would be good to have agreement that
+includes input from these past discussions as well as the wisdom of the
+community. The best way to do this is to [file an
+issue](https://github.com/CCI-MOC/ims/issues) on github, or speak with one of the core developers directly.
+
+
+#### Testing
+
+* [Testing document](testing.md)
+
+Testing helps to ensure the quality of the code base. Every pull request
+submitted should first be tested to ensure that all existing tests pass.
+
+When introducing new functionality, new tests (both unit and more comprehensive)
+should be added that provide adequate coverage.
+
+If fixing a bug, a regression test should accompany the bug fix to ensure that
+the bug does not return.
+
+Once the checklist above has been met, open a Pull Request from your
+personal fork to the main BMI repo.
+
+Checkout the [testing document](testing.md) for more details.
+
+
+### Code review
+
+Code reviews help increase code quality by finding:
+* mistakes that can oftentimes be overlooked by a single developer
+* improving readability
+* helping reviewers to learn about different areas of the code
+
+Reviewers are the final guardians for good code quality. If you see a piece of
+complex code, that is probably where you want to spend a lot of your review
+time. Remember the 80/20 rule (80% of the bugs come from 20% of the code).
+
+
+### Approval criteria on github
+
+Pull Requests require at least 2 approvals, also known as "+1"'s, in
+order to be merged, with at least 2 core developers being involved in some way:
+either as a submitter or a reviewer.
+
+The exception to this is documentation, where we are a little more lenient in
+the interest of lowering the barrier to having better docs. For docs changes,
+typically one +1 is sufficient.
+
+Whomever provides the enabling +1 is responsible for clicking the merge button
+on github. If you do not have commit access, then please add your +1 to the PR
+and ask one of the Core Developers to complete the merge.
```

Comment:
Even this is not followed. We are asking whoever is the team lead before merging.